app/src/main/java/fr/free/nrw/commons/Media.java

```diff
@@ -164,9 +168,16 @@ public static Media from(MwQueryPage page) {
      */
     private static String getArtist(ExtMetadata metadata) {
         try {
+            final String anchorStartTagTerminalChars = "\">";
+            final String anchorCloseTag = "</a>";
+
             String artistHtml = metadata.artist();
-            return artistHtml.substring(artistHtml.indexOf("title=\""), artistHtml.indexOf("\">"))
-                    .replace("title=\"User:", "");
+            return artistHtml
+                .substring(
+                    artistHtml.indexOf(anchorStartTagTerminalChars),
+                    artistHtml.indexOf(anchorCloseTag)
+                )
+                .substring(anchorStartTagTerminalChars.length());
```

Comment:
We get a string of 2 characters after getting the contents of the anchor tag? That doesn't seem like an artist to me, just from reading the code

Reply:
No actually it is string starting from index 2. this formats  "> these 2 characters.

Reply:
D'oh, yeah. I am so unused to the awful java substring apis after using the fantastic kotlin ones for so long. Maybe add the length on to the index call I think that flows better

Reply:
Or I guess we could use the Kotlin stdLib...
![image](https://user-images.githubusercontent.com/3358282/78874068-b2d62e80-7a43-11ea-8921-9aa3f20d4d7a.png)

Blugh, you have to specify the default value, ugly.

Reply:
This default (">asdf) will look more ugly or set null or username is set as default.
I think it's better to leave it in java

Reply:
Yeah keep it in java, I was just idly musing.

>Maybe add the length on to the index call I think that flows better

This however I would like to see